app/src/main/java/com/example/campusguide/Accessibility.kt

```diff
@@ -0,0 +1,49 @@
+package com.example.campusguide
+
+/**
+ * Class handles accessibility requests such as adjustments
+ * to directions and/or colors of the application.
+ */
+object Accessibility {
+
+    var forbiddenRooms: MutableList<String> = emptyList<String>().toMutableList()
+
+    // public methods to be called on click
+    fun setEscalators(bool: Boolean) {
+        if (!bool) forbidEscalators()
+        else allowEscalators()
+    }
+
+    fun setElevators(bool: Boolean) {
+        if (!bool) forbidElevators()
+        else allowElevators()
+    }
+
+    fun setStairs(bool: Boolean) {
+        if (!bool) forbidStairs()
+        else allowStairs()
+    }
+
+    // private methods that change pathfinding
+    private fun forbidEscalators() {
+        forbiddenRooms.add("escalators")
+    }
+
+    private fun forbidElevators() {
+        forbiddenRooms.add("elevators")
```

Comment:
You can make strings `"elevators"`, `"stairs"` and `"elevators"` as string variables to pass.